staging/src/kubevirt.io/client-go/apis/flavor/v1alpha1/types.go

```diff
@@ -0,0 +1,84 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// VirtualMachineFlavor - TODO -- add description
+//
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+// +genclient
+type VirtualMachineFlavor struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec VirtualMachineFlavorSpec `json:"spec" valid:"required"`
+}
+
+// VirtualMachineFlavorList - TODO -- add description
+//
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+type VirtualMachineFlavorList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []VirtualMachineFlavor `json:"items"`
+}
+
+// VirtualMachineClusterFlavor - TODO -- add description
+//
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+// +genclient
+// +genclient:nonNamespaced
+type VirtualMachineClusterFlavor struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec VirtualMachineFlavorSpec `json:"spec" valid:"required"`
+}
+
+// VirtualMachineClusterFlavorList - TODO -- add description
+//
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+type VirtualMachineClusterFlavorList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []VirtualMachineClusterFlavor `json:"items"`
+}
+
+// +k8s:openapi-gen=true
+type VirtualMachineFlavorSpec struct {
+	DefaultProfile VirtualMachineFlavorProfile         `json:"defaultProfile"`
+	CustomProfiles []VirtualMachineFlavorCustomProfile `json:"customProfiles,omitempty"`
+}
+
+// +k8s:openapi-gen=true
+type VirtualMachineFlavorProfile struct {
+	CpuCount *uint32 `json:"cpuCount,omitempty" optional:"true"`
```

Comment:
I think we'll want this to be a struct.  It's probably as simple as this being the Domain's CPU struct. 

```CPU *CPU `json:"cpu,omitempty" optional:"true"` ```

Reply:
Yes, this API was just a placeholder. I will use the `CPU` struct.